### PR TITLE
Make carriage returns illegal

### PR DESCRIPTION
**Pull request by kantai**
This makes carriage returns illegal #1118

### CODE REVIEW

File: src/vm/ast/parser/mod.rs [hunk 1]
```diff
@@ -68,7 +68,7 @@ pub fn lex(input: &str) -> ParseResult<Vec<(LexItem, u32, u32)>> {
     let lex_matchers: &[LexMatcher] = &[
         LexMatcher::new(r##""(?P<value>((\\")|([[:print:]&&[^"\n\r\t]]))*)""##, TokenType::StringLiteral),
```
Comment (lgalabru) [inline src/vm/ast/parser/mod.rs]: Should we also update this regex?
`\n\r\t` to `\n\t`

File: src/vm/ast/parser/mod.rs [hunk 2]
```diff
@@ -68,7 +68,7 @@ pub fn lex(input: &str) -> ParseResult<Vec<(LexItem, u32, u32)>> {
...
         LexMatcher::new(";;[[:print:]&&[^\n\r]]*", TokenType::Whitespace), // ;; comments.
```
Comment (lgalabru) [inline src/vm/ast/parser/mod.rs]: And this one?
`\n\r` to `\n`

Comment (kantai) [reply src/vm/ast/parser/mod.rs]: It shouldn't matter in either of these, though we should maybe clean up the regex (the `:print:` class is equivalent to `[ -~]`, which doesn't include any of `\n` `\r` or `\t`

Comment (kantai) [reply src/vm/ast/parser/mod.rs]: Addressed in: cab43a12a5a4395272d37c43d68610f47939aa5f

Comment (jcnelson) [reply src/vm/ast/parser/mod.rs]: Wait, I thought we wanted `\t` and `\n` as valid whitespace characters?  We at least want `\n`...

File: src/vm/ast/parser/mod.rs (hunk 1 above)
Comment (kantai) [reply src/vm/ast/parser/mod.rs]: Addressed in: cab43a12a5a4395272d37c43d68610f47939aa5f

File: src/vm/ast/parser/mod.rs (hunk 2 above)
Comment (kantai) [reply src/vm/ast/parser/mod.rs]: that's a separate whitespace class

Comment (kantai) [reply src/vm/ast/parser/mod.rs]: (this one is for comments)

**Comment by lgalabru** [comment]
If this PR is also addressing #1115, do you think we can also have some tests for testing the **Case 2** of the exploit?


Screen Shot 2019-11-19 at 5 38 48 PM
 of the exploit?

**Comment by kantai** [comment]
Added test cases for CR without LF, NEL, PS, LS, type line breaks. All of them are forbidden and correctly lead to parse errors.

**Pull request merged**